quiz/test_session.py

```diff
@@ -2,57 +2,56 @@
 
     def test_given_new_session__has_no_correct_answers(self):
-        session = {}
-        s = UserData(session)
-        self.assertSetEqual(set(), s.get_correctly_answered_questions())
+        user_data = create_session_with_answers_to(tuple())
+        self.assertSetEqual(set(), user_data.get_correctly_answered_questions())
 
     def test_given_new_session__after_a_correct_answer_to_a_published_question_is_registered__it_is_listed_as_answered(self):
-        q1 = Question.objects.create(state='PUB', difficulty=1)
-        q2 = Question.objects.create(state='PUB', difficulty=1)
-        session = {}
-        s = UserData(session)
-        s.register_correct_answer(q1.pk)
-        self.assertSetEqual({q1.pk}, s.get_correctly_answered_questions())
-        s.register_correct_answer(q2.pk)
-        self.assertSetEqual({q1.pk, q2.pk}, s.get_correctly_answered_questions())
+        [q1, q2] = create_questions(2, state='PUB')
+        user_data = create_session_with_answers_to((q1,))
+        self.assertSetEqual({q1.pk}, user_data.get_correctly_answered_questions())
+        user_data.register_correct_answer(q2.pk)
+        self.assertSetEqual({q1.pk, q2.pk}, user_data.get_correctly_answered_questions())
 
     def test_given_new_session__after_a_correct_answer_to_a_retracted_question_is_registered__it_is_not_listed_as_answered(self):
-        q = Question.objects.create(state='RET', difficulty=1)
-        session = {}
-        s = UserData(session)
-        s.register_correct_answer(q.pk)
-        self.assertSetEqual(set(), s.get_correctly_answered_questions())
+        [q] = create_questions(1, state='RET')
+        user_data = create_session_with_answers_to((q,))
+        self.assertSetEqual(set(), user_data.get_correctly_answered_questions())
 
     def test_given_new_session__after_a_published_question_becomes_retracted__its_answer_is_not_listed_as_answered(self):
-        q = Question.objects.create(state='PUB', difficulty=1)
-        session = {}
-        s = UserData(session)
-        s.register_correct_answer(q.pk)
-        self.assertSetEqual({q.pk}, s.get_correctly_answered_questions())
+        [q] = create_questions(1, state='PUB')
+        user_data = create_session_with_answers_to((q,))
+        self.assertSetEqual({q.pk}, user_data.get_correctly_answered_questions())
         q.state = 'RET'
         q.save()
-        self.assertSetEqual(set(), s.get_correctly_answered_questions())
+        self.assertSetEqual(set(), user_data.get_correctly_answered_questions())
 
     def test_given_existing_session__correct_answers_are_still_there(self):
-        q = Question.objects.create(state='PUB', difficulty=1)
-        old_data = UserData({})
-        old_data.register_correct_answer(q.pk)
-        session = {'user_data': old_data}
-        new_data = UserData(session)
+        [q] = create_questions(1)
+        old_data = create_session_with_answers_to((q,))
+        new_data = UserData({'user_data': old_data})
```

Comment:
I think I'd keep the `session` temporary variable, to show more clearly that we're simulating creating `UserData` from a session.